google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/internal/RowSetUtilTest.java

```diff
@@ -0,0 +1,573 @@
+            .addRowKeys(ByteString.copyFromUtf8("d"))
+            .build();
+
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startClosed("a").endClosed("d"));
+  }
+
+  @Test
+  public void singleRange1BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(
+                RowRange.newBuilder()
+                    .setStartKeyClosed(ByteString.copyFromUtf8("a"))
+                    .setEndKeyClosed(ByteString.copyFromUtf8("b")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startClosed("a").endClosed("b"));
+  }
+
+  @Test
+  public void singleRange2BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(
+                RowRange.newBuilder()
+                    .setStartKeyClosed(ByteString.copyFromUtf8("a"))
+                    .setEndKeyOpen(ByteString.copyFromUtf8("b")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startClosed("a").endOpen("b"));
+  }
+
+  @Test
+  public void singleRange3BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(
+                RowRange.newBuilder()
+                    .setStartKeyOpen(ByteString.copyFromUtf8("a"))
+                    .setEndKeyOpen(ByteString.copyFromUtf8("b")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startOpen("a").endOpen("b"));
+  }
+
+  @Test
+  public void singleRange4BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(
+                RowRange.newBuilder()
+                    .setStartKeyOpen(ByteString.copyFromUtf8("a"))
+                    .setEndKeyClosed(ByteString.copyFromUtf8("b")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startOpen("a").endClosed("b"));
+  }
+
+  @Test
+  public void singleRangeUnbounded1BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(RowRange.newBuilder().setStartKeyClosed(ByteString.copyFromUtf8("a")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().startClosed("a"));
+  }
+
+  @Test
+  public void singleRangeUnbounded2BoundTest() {
+    RowSet rowSet =
+        RowSet.newBuilder()
+            .addRowRanges(RowRange.newBuilder().setEndKeyClosed(ByteString.copyFromUtf8("z")))
+            .build();
+    ByteStringRange actual = RowSetUtil.getBound(rowSet);
+    assertThat(actual).isEqualTo(ByteStringRange.unbounded().endClosed("z"));
+  }
+
+  @Test
+  public void multipleRangeBoundTest() {
```

Comment:
What about multipleRange overlapped and mixed of row keys and ranges?

Reply:
done